xarray/core/utils.py

```diff
@@ -594,19 +594,24 @@ def __len__(self):
 
 
 def to_numeric(array, offset=None, datetime_unit=None, dtype=float):
-    """
-    Make datetime array float
+    """Convert an array containing datetime-like data to an array of floats.
```

Comment:
It sounds like this function do nothing to numeric arrays. Maybe the default offset should be zero and we can pass the minimum for the datetime array manually.

Reply:
Ah thanks to your earlier comment I see where you are coming from now (my initial thought was that in all cases we would make sure an array contained dates before calling `to_numeric` on it).  Depending on what we decide to do there I can modify this function as needed.

Reply:
I just imagined if we use this function for different purpose in the future. I have not a particular idea yet but i thought we may not always want to subtract minimum.

Reply:
For now I renamed `to_numeric` to `datetime_to_numeric` to make it clear that it is intended only to be used with datetimes (I've tried to make sure now that this is always the case).  Does that seem reasonable?  I think we can always change the defaults to this internal utility function if we find a need later.